addons/hw_drivers/tools/wifi.py

```diff
@@ -0,0 +1,278 @@
+
+    # Copy the configuration file to the root filesystem
+    command = ['sudo', 'cp', source_path, destination_path]
+
+    if forget_network and destination_path.exists():
+        # Remove the configuration file from the root filesystem
+        command = ['sudo', 'rm', '-f', destination_path]
+
+    with writable():
+        if subprocess.run(command, check=False).returncode == 0:
+            return True
+        else:
+            _logger.error('Failed to apply the network configuration to root_bypass_ramdisks.')
+            return False
+
+
+# -------------------------- #
+# Access Point Configuration #
+# -------------------------- #
+
+@cache
+def get_access_point_ssid():
+    """Generate a unique SSID for the access point.
+    Uses the MAC address of the device without the colons, or
+    a random token if mac was not found.
+
+    :return: Generated SSID
+    :rtype: str
+    """
+    mac = get_mac_address()
+    return "IoTBox-" + (''.join(mac.split(':')) if mac else secrets.token_hex(6))
+
+
+def _configure_access_point(on=True):
+    """Update the ``hostapd`` configuration file with the given SSID.
+    This method also adds/deletes a static IP address to the ``wlan0`` interface,
+    mandatory to allow people to connect to the access point.
+
+    :param bool on: Start or stop the access point
+    :return: True if the configuration is successful
+    """
+    ssid = get_access_point_ssid()
+
+    _logger.info("Configuring access point with SSID %s", ssid)
+    with writable():
+        if on:
+            with open('/etc/hostapd/hostapd.conf', 'w', encoding='utf-8') as f:
+                f.write(f"interface=wlan0\nssid={ssid}\nchannel=1\n")
+    mode = 'add' if on else 'del'
+    return (
+        subprocess.run(['sudo', 'ip', 'address', mode, '10.11.12.1/24', 'dev', 'wlan0'], check=False).returncode == 0
+        or not on  # Don't fail if stopping access point: IP address might not exist
+    )
+
+
+def toggle_access_point(state=START):
+    """Start or stop an access point.
+
+    :param bool state: Start or stop the access point
+    :return: True if the operation on the access point is successful
+    :rtype: bool
+    """
+    if not _configure_access_point(state):
+        return False
+
+    _logger.info("Starting access point.")
+    mode = 'start' if state else 'stop'
+    if subprocess.run(['sudo', 'systemctl', mode, 'hostapd'], check=False).returncode == 0:
+        return True
+    else:
+        _logger.error("Failed to start access point.")
+        return False
+
+
+def is_access_point():
+    """Check if the device is currently in access point mode.
+
+    :return: True if the device is in access point mode
+    :rtype: bool
+    """
+    # We only get the first line as `loopback` can still be active when `wlan0` is up
+    device = _nmcli(['-g', 'device', 'connection', 'show', '--active'], sudo=True).splitlines()[0]
+    return device == 'loopback'
```

Comment:
This is always returning `False` for me (testing on RPi 5). When I run the command manually it returns `lo` not `loopback`.

Reply:
ah yes you're right, sorry